beta_carrier_metadata.go

```diff
@@ -0,0 +1,99 @@
+package easypost
+
+import (
+	"context"
+	"strings"
+)
+
+// BetaCarrierMetadata represents all the metadata for a carrier.
+type BetaCarrierMetadata struct {
+	Name               string                       `json:"name,omitempty"`
+	HumanReadable      string                       `json:"human_readable,omitempty"`
+	ServiceLevels      []*BetaMetadataServiceLevel      `json:"service_levels,omitempty"`
+	PredefinedPackages []*BetaMetadataPredefinedPackage `json:"predefined_packages,omitempty"`
+	ShipmentOptions    []*BetaMetadataShipmentOption    `json:"shipment_options,omitempty"`
+	SupportedFeatures  []*BetaMetadataSupportedFeature  `json:"supported_features,omitempty"`
+}
+
+// BetaMetadataServiceLevel represents an available service level of a carrier.
+type BetaMetadataServiceLevel struct {
+	Name          string   `json:"name,omitempty"`
+	Carrier       string   `json:"carrier,omitempty"`
+	HumanReadable string   `json:"human_readable,omitempty"`
+	Description   string   `json:"description,omitempty"`
+	Dimensions    []string `json:"dimensions,omitempty"`
+	MaxWeight     float64  `json:"max_weight,omitempty"`
+}
+
+// BetaMetadataPredefinedPackage represents an available predefined package of a carrier.
+type BetaMetadataPredefinedPackage struct {
+	Name          string   `json:"name,omitempty"`
+	Carrier       string   `json:"carrier,omitempty"`
+	HumanReadable string   `json:"human_readable,omitempty"`
+	Description   string   `json:"description,omitempty"`
+	Dimensions    []string `json:"dimensions,omitempty"`
+	MaxWeight     float64  `json:"max_weight,omitempty"`
+}
+
+// BetaMetadataShipmentOption represents an available shipment option of a carrier.
+type BetaMetadataShipmentOption struct {
+	Name          string `json:"name,omitempty"`
+	Carrier       string `json:"carrier,omitempty"`
+	HumanReadable string `json:"human_readable,omitempty"`
+	Description   string `json:"description,omitempty"`
+	Deprecated    bool   `json:"deprecated,omitempty"`
+	Type          string `json:"type,omitempty"`
+}
+
+// BetaMetadataSupportedFeature represents a supported feature of a carrier.
+type BetaMetadataSupportedFeature struct {
+	Name        string `json:"name,omitempty"`
+	Carrier     string `json:"carrier,omitempty"`
+	Description string `json:"description,omitempty"`
+	Supported   bool   `json:"supported,omitempty"`
+}
+
+// BetaGetCarrierMetadata retrieves all metadata for all carriers on the EasyPost platform.
+func (c *Client) BetaGetCarrierMetadata() (out []*BetaCarrierMetadata, err error) {
```

Comment:
Question: do we want to make these functions deprecated in Go?

Source I was looking at: https://stackoverflow.com/questions/7849663/how-do-you-mark-code-as-deprecated-in-go

Reply:
Yes, we should mark them as deprecated

Reply:
Fixed